backend/core/llm_chain.py

```
@@ -3,490 +3,317 @@
+    logger.info(f"Query routed to retrievers: {[name for name in selected_names if name in retrievers]}")
+    return selected_retrievers
 
-        response = rag_chain.invoke({
-            "input": user_input,
-            "chat_history": chat_history
-        })
 
-        return response.get("answer", "I'm sorry, I couldn't generate a response.")
+def get_cached_retrieval(cache_key: str) -> Optional[List[Document]]:
+    """Checks the retrieval cache for a list of documents."""
+    if not cache_key or not ENABLE_CACHING: return None
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)